bibliography.bib

```diff
@@ -7,6 +7,68 @@ @inproceedings{gullasch2011cache
+  booktitle={Journal of Cryptology},
+  year={2009},
+}
+
+@incollection{ristenpart2009cloud,
+  title={Hey, you, get off of my cloud: exploring information leakage in third-party compute clouds},
+  author={Ristenpart, Thomas and Tromer, Eran and Shacham, Hovav and Savage, Stefan},
+  booktitle={CCS '09 Proceedings of the 16th ACM conference on Computer and communications security},
+  pages={199--212},
+  year={2009},
+  publisher={ACM},
+  annote={
+    \setlength{\parskip}{1.5ex}
+    Describes a method for achieving co-residence with target VMs on the cloud and exploiting information leakage between these VMs using 
+    Amazon's EC2 service as an example.
+    The attack takes advantage of the fact that many VMs can be created by the same user, and that there is a possibility that these might 
+    run on the same hardware as a target due to the way the VMs share physical resources.
+    Once co-location is achieved, an attacker can use side-channels (namely caches) to get information about the other instances running on 
+    the same machine as the malicious VM.  
+    \par
+    A 'map' of the EC2 service was constructed in order improve the likelihood of placing a VM on the same host as the target. 
+    By creating multiple instances for each permutation of the creation parameters, the authors observed a trend between internal IP addresses
+    and creation parameters. 
+    These internal IP addresses were concluded to be statically associated with physical machines. 
+    Use of the created map requires translating the external IP of the target to an internal IP, which is made possible by the DNS service
+    provided by EC2.
+    \par
+    There are multiple ways to check if two instances are co-resident. 
+    Each physical machine has a privileged VM associated with it called Domain0, or Dom0.
+    Two VMs which are co-located will have identical Dom0 IP addresses and internal IP addresses which are numerically close.
+    Measuring the round-trip time for a packet sent between the two VMs can also be used to tell if they are co-resident, as it is 
+    significantly lower in co-resident instances.
+    The Dom0 IP check was used for all subsequent co-location testing as experiments showed that it produced no false positives.
+    \par
+    Two strategies were used to prove that achieving co-residence was possible. 
+    The first used a brute-force technique by creating a large number of probe instances and targeting a large number of potential victims.
+    The probes checked if they were co-resident with any of the target VMs, then were immediately deactivated if they were not.
+    This method achieved a reasonable success rate, covering 8.4\% of the targets. 
+    A second method monitored a victim server, waiting for it to disappear when not needed, then reappear as a new instance. 
+    The attacker then ran many instances using the same creation parameters as the target. 
```

Comment:
Still want to know how an attacker discovers these parameters in practice. Does the paper mention that problem at all?